src/types/currency.ts

```diff
@@ -1,93 +1,133 @@
+ */
+function isHex(hex: string): boolean {
+  return HEX_REGEX.test(hex);
 }
 
-function isStringRepr(val) {
-  return _.isString(val) && (isISOCode(val) || isHex(val));
+/**
+ * Tests if a string is a valid representation of a currency
+ */
+function isStringRepresentation(str: string): boolean {
+  return isISOCode(str) || isHex(str);
 }
 
-function isBytesArray(val) {
-  return val.length === 20;
+/**
+ * Tests if a Buffer is a valid representation of a currency
+ */
+function isBytesArray(bytes: Buffer): boolean {
+  return bytes.byteLength === 20;
 }
 
-function isValidRepr(val) {
-  return isStringRepr(val) || isBytesArray(val);
+/**
+ * Ensures that a value is a valid representation of a currency
+ */
+function isValidRepresentation(val: Buffer | string): boolean {
+  return val instanceof Buffer
+    ? isBytesArray(val)
+    : isStringRepresentation(val);
 }
 
-function bytesFromRepr(val) {
-  if (isValidRepr(val)) {
-    // We assume at this point that we have an object with a length, either 3,
-    // 20 or 40.
-    return val.length === 3 ? isoToBytes(val) : val;
+/**
+ * Generate bytes from a string or buffer representation of a currency
+ */
+function bytesFromRepresentation(val: string): Buffer {
+  if (!isValidRepresentation(val)) {
+    throw new Error(`Unsupported Currency representation: ${val}`);
   }
-  throw new Error(`Unsupported Currency repr: ${val}`);
+  return val.length === 3 ? isoToBytes(val) : Buffer.from(val, "hex");
 }
 
-const $uper = Hash160.prototype;
-const Currency = makeClass(
-  {
-    inherits: Hash160,
-    getters: ["isNative", "iso"],
-    statics: {
-      init() {
-        this.XRP = new this(new Uint8Array(20));
-      },
-      from(val) {
-        return val instanceof this ? val : new this(bytesFromRepr(val));
-      },
-    },
-    Currency(bytes) {
-      Hash160.call(this, bytes);
-      this.classify();
-    },
-    classify() {
-      // We only have a non null iso() property available if the currency can be
-      // losslessly represented by the 3 letter iso code. If none is available a
-      // hex encoding of the full 20 bytes is the canonical representation.
-      let onlyISO = true;
-
-      const bytes = this._bytes;
-      const code = slice(this._bytes, 12, 15, Array);
-      const iso = code.map((c) => String.fromCharCode(c)).join("");
-
-      for (let i = bytes.length - 1; i >= 0; i--) {
-        if (bytes[i] !== 0 && !(i === 12 || i === 13 || i === 14)) {
-          onlyISO = false;
-          break;
-        }
-      }
-      const lossLessISO = onlyISO && iso !== "XRP" && ISO_REGEX.test(iso);
-      this._isNative = onlyISO && _.isEqual(code, [0, 0, 0]);
-      this._iso = this._isNative ? "XRP" : lossLessISO ? iso : null;
-    },
-    toJSON() {
-      if (this.iso()) {
-        return this.iso();
+/**
+ * Class defining how to encode and decode Currencies
+ */
+class Currency extends Hash160 {
+  static XRP = new Currency(Buffer.alloc(20));
```

Comment:
Can we mark these as private?

Reply:
No, `Currency.XRP` is also used in class `Amount`.